### PR TITLE
Add CreatePost component

### PR DESCRIPTION
**Pull request by MatYouKy**
# Related issue
- Create Post

# Changes
- Created custom hook to adding posts,
- Add styles to CreatePost component,
- Add testing component,

# Additional information
- none

# Checklist
- [X] Performed self-review
- [X] Added data-testId for UI elements
- [X] Tests
- [X] Storybook

# Screenshots

- Add screenshots

 CreatePost

CreatePost_1

### CODE REVIEW

File: src/components/CreatePost/CreatePost.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      timestamp: generateDate(),
+      avatar: currentUser?.photoURL
+    });
+
+    setPostText('');
+  };
+
+  return (
+    <article className="create-post">
+      <Avatar className="avatar" />
```
Comment (jakub-gosciniak) [inline src/components/CreatePost/CreatePost.tsx:41]: add src to avatar to display current user's avatar from currentUser?.photoURL

File: src/actions/setPost.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+interface FeedPostProps {
+  ownerUid: string | undefined | null;
+  displayName: string | undefined | null;
+  avatar?: string | undefined | null;
+  content: string;
+  timestamp: string;
+}
+
+export const setPost = async ({ ownerUid, content, avatar, timestamp, displayName }: FeedPostProps): Promise<void> => {
```
Comment (mateusz0612) [inline src/actions/setPost.tsx:11]: I think you don't need to use async if you are not using await in your function.

File: src/utilities/date.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  const date = new Date();
+  const year = date.getFullYear();
+  const month = date.getMonth() + 1 < 10 ? `0${date.getMonth()}` : date.getMonth();
+  const day = date.getDate();
+  const hours = date.getHours();
+  const minutes = date.getMinutes() < 10 ? `0${date.getMinutes()}` : date.getMinutes();
+  const seconds = date.getSeconds();
+
+  return `${year}-${month}-${day} ${hours}:${minutes}:${seconds}`;
+};
```
Comment (Polo11121) [inline src/utilities/date.ts:11]: It looks strange, surely there is no better way to implement it?

**Review by Polo11121 — DISMISSED** [review]

File: src/utilities/date.ts (hunk 3 above)
Comment (mateusz0612) [reply src/utilities/date.ts:11]: Since we are using moment in our project you can simply do moment().format("YYYY-MM-D h:m:s")

Comment (MateuszLisowski) [reply src/utilities/date.ts:11]: probably it's the best idea

File: src/actions/setPost.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { db } from '../firebase';
+
+interface FeedPostProps {
+  ownerUid: string | undefined | null;
+  displayName: string | undefined | null;
+  avatar?: string | undefined | null;
+  content: string;
+  timestamp: string;
```
Comment (MateuszLisowski) [inline src/actions/setPost.tsx:8]: it's not date?

File: src/actions/setPost.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { db } from '../firebase';
+
+interface FeedPostProps {
+  ownerUid: string | undefined | null;
+  displayName: string | undefined | null;
```
Comment (MateuszLisowski) [inline src/actions/setPost.tsx:5]: it could be either null or undefined?? it's weird - it refers to the three properties

**Review by MateuszLisowski — CHANGES_REQUESTED** [review]
I agree with other reviewers

File: src/actions/setPost.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  avatar?: string | undefined | null;
+  content: string;
+  timestamp: string;
+}
+
+export const setPost = async ({ ownerUid, content, avatar, timestamp, displayName }: FeedPostProps): Promise<void> => {
+  db.collection('posts')
+    .add({ ownerUid, content, avatar, timestamp, displayName })
+    .then(data => console.log(data))
```
Comment (MateuszLisowski) [inline src/actions/setPost.tsx:14]: why console.log data??

**Review by MateuszLisowski — APPROVED** [review]

**Review by mateusz0612 — APPROVED** [review]

**Review by jakub-gosciniak — APPROVED** [review]

**Review by Polo11121 — APPROVED** [review]

**Pull request merged**